src/hooks/useKeyboardNav.tsx

```diff
@@ -0,0 +1,22 @@
+import {useState} from "react";
+import {ISearchingResult} from "../types/types";
+
+export const useKeyboardNav = (searchingResults: ISearchingResult[], updateData: (title: string) => void): [React.KeyboardEventHandler<HTMLDivElement>, number] => {
```

Comment:
Here we have hook that was created for general purposes but we added custom type so it can't be reused for other types.
Try to use generic type for incoming value and for updateData. In this case updateData should accept whole item instead of title as it should not no details of this logic.